Add content to ObjectManager topic (#4699)
magento/devdocs: guides/v2.1/extension-dev-guide/object-manager.md

---
group: php-developer-guide
title: ObjectManager

> menu_title: ObjectManager

Reviewer: menu_title can be deleted too.

You can depend on and use the `ObjectManager` class in the following scenarios:

- You can use the object manager in static magic methods like `__wakeup()`, `__sleep()`, etc.
  - An example can be found in the `__wakeup()` method in the [`Magento/Eav/Model/Entity/Attribute/AbstractAttribute`][] class.
- You can use the `ObjectManager` to maintain backward compatibility for a constructor.
- In a global scope, like in fixtures of integration tests, you can use the object manager.

> - The object manager can be a dependency in classes used for the creation of objects, e.g. factories or proxies.

Reviewer: ```suggestion
- The object manager can be a dependency in classes used for creating objects such as factories or proxies.
```
